### DISCUSSION TITLE
Any guide to add new display

### DISCUSSION THREAD

**Discussion by allenchak** [olikraus/u8g2#2059] [General]
I have a GLCD display which is 128x64, similar to others. I would like to help to merge my LM6023 (using S1D15300 IC) repo to yours, may I know any guide / reference for adding a new.

Or just let me know if I provide anything could assist you to add it quickly, such as the initial commands, power control commands, etc.

**Comment by allenchak** [comment]
```
display_info =
    {
        /* chip_enable_level = */ 0,
        /* chip_disable_level = */ 1,

        /* post_chip_enable_wait_ns = */ 400, /* st7565 datasheet, table 26, tcsh */
        /* pre_chip_disable_wait_ns = */ 120,  /* st7565 datasheet, table 26, tcss */
        /* reset_pulse_width_ms = */ 1,
        /* post_reset_wait_ms = */ 1,
        /* sda_setup_time_ns = */ 200,   /* st7565 datasheet, table 26, tsds */
        /* sck_pulse_width_ns = */ 1000, /* half of cycle time (100ns according to datasheet), AVR: below 70: 8 MHz, >= 70 --> 4MHz clock */
        /* sck_clock_hz = */ 1000000UL, /* since Arduino 1.6.0, the SPI bus speed in Hz. Should be  1000000000/sck_pulse_width_ns */
        /* spi_mode = */ 0,             /* active high, rising edge */
        /* i2c_bus_clock_100kHz = */ 4,
        /* data_setup_time_ns = */ 200,   /* st7565 datasheet, table 24, tds8 */
        /* write_pulse_width_ns = */ 200, /* st7565 datasheet, table 24, tcclw */
        /* tile_width = */ 16,           /* width of 16*8=128 pixel */
        /* tile_hight = */ 8,
        /* default_x_offset = */ 0,
        /* flipmode_x_offset = */ 0,
        /* pixel_width = */ 128,
        /* pixel_height = */ 64};
```

```

init_seq{


    U8X8_START_TRANSFER(), /* enable chip, delay is part of the transfer start */

    U8X8_C(0x0e2), /* soft reset */

    U8X8_C(0x0ae), /* display off */
    U8X8_C(0x040), /* set display start line to 0 */
    U8X8_C(0x0a0), /* ADC set to normal */
    U8X8_C(0x0a6), /* Display normal */
    U8X8_C(0x0a4), /* all point normal */

    U8X8_C(0x0a2), /* LCD bias 1/8 */
    U8X8_C(0x0c0), /* common output mode */
    U8X8_C(0x02f), /* all power control circuits on */


    U8X8_C(0x090), /* set contrast */
    U8X8_C(0x0af), /* display on */

    U8X8_C(0x0b0), /* set page address */
    U8X8_C(0x01f), /* set column address upper */
    U8X8_C(0x000), /* set column address lower */

    U8X8_END_TRANSFER(), /* disable chip */
    U8X8_END()           /* end of sequence */
}

```

**Reply by olikraus** [reply]
> I would like to help to merge my LM6023 (using S1D15300 IC) repo

Thanks. There is some postprocessing required, so let's do it together.

Do you have already modified an existing constructor? 
Which constructor did you use as template?

**Comment by allenchak** [comment]
I used ST7565 and tested with replaced upper value to U8G2_ST7565_JLX12864_2_4W_SW_SPI

**Reply by olikraus** [reply]
How does it look (picture)? Can use flip mode example to test your changes? There should be a solid frame all the time

**Reply by allenchak** [reply]
[LM6023 Graphics test](https://youtu.be/H1a-3v-HCV4) 

[LM6023 Flip Mode test](https://youtu.be/J1bIqsKD_Gg)

**Reply by olikraus** [reply]
Looks like F0 mode is not correct. One more question: Did you also test the `U8G2_ST7565_LM6063_2_4W_SW_SPI` constructor? If not can you do so?

**Reply by allenchak** [reply]
Nothing output, the screen is blank

**Reply by olikraus** [reply]
ok, then, I assume this is the datasheet for the controller: https://github.com/allenchak/LM6023/blob/master/Datasheet/S1D15300.pdf

I also assume, that this is the datasheet for the display: https://github.com/allenchak/LM6023/blob/master/Datasheet/SP-MCU-20110107-01_LM6023A-Manual-Rev0.3.pdf

right?

**Comment by olikraus** [comment]
Created issue, let's continue here: https://github.com/olikraus/u8g2/issues/2063
